modules/base/rendering/renderableprism.cpp

```diff
@@ -0,0 +1,354 @@
+    return _shader != nullptr;
+}
+
+void RenderablePrism::initialize() {
+    updateVertexData();
+}
+
+void RenderablePrism::initializeGL() {
+    initializeShader();
+    ghoul::opengl::updateUniformLocations(*_shader, _uniformCache, UniformNames);
+
+    glGenVertexArrays(1, &_vaoId);
+    glGenBuffers(1, &_vboId);
+    glGenBuffers(1, &_iboId);
+
+    glBindVertexArray(_vaoId);
+
+    updateBufferData();
+
+    glEnableVertexAttribArray(_locVertex);
+    glVertexAttribPointer(_locVertex, 3, GL_FLOAT, GL_FALSE, 3 * sizeof(float), nullptr);
+    glBindVertexArray(0);
+}
+
+void RenderablePrism::deinitializeGL() {
+    global::renderEngine->removeRenderProgram(_shader.get());
+    _shader = nullptr;
+
+    glDeleteVertexArrays(1, &_vaoId);
+    _vaoId = 0;
+
+    glDeleteBuffers(1, &_vboId);
+    _vboId = 0;
+
+    glDeleteBuffers(1, &_iboId);
+    _iboId = 0;
+}
+
+void RenderablePrism::updateVertexData() {
+    _vertexArray.clear();
+    _indexArray.clear();
+
+    // Get unit circle vertices on the XY-plane
+    std::vector<float> unitVertices;
+    std::vector<float> unitVerticesLines;
+    unitVertices.reserve(2 * _nShapeSegments);
+    unitVerticesLines.reserve(2 * _nLines);
+    getUnitCircleVertices(unitVertices, _nShapeSegments);
+    getUnitCircleVertices(unitVerticesLines, _nLines);
+
+    // Put base and top shape vertices into array
+    for (int i = 0; i < 2; ++i) {
+        float h = i * _length; // z value, 0 to _length
+
+        for (int j = 0, k = 0; j < _nShapeSegments && k < unitVertices.size(); ++j) {
+            float ux = unitVertices[k++];
+            float uy = unitVertices[k++];
+
+            _vertexArray.push_back(ux * _radius); // x
+            _vertexArray.push_back(uy * _radius); // y
+            _vertexArray.push_back(h);            // z
+        }
+    }
+
+    // Put the vertices for the connecting lines  into array
+    if (_nLines == 1) {
+        // In the case of just one line then connect the center points instead
+        // Center for base shape
+        _vertexArray.push_back(0.f);
+        _vertexArray.push_back(0.f);
+        _vertexArray.push_back(0.f);
+
+        // Center for top shape
+        _vertexArray.push_back(0.f);
+        _vertexArray.push_back(0.f);
+        _vertexArray.push_back(_length);
+    }
+    else {
+        for (int j = 0, k = 0; j < _nLines && k < unitVerticesLines.size(); ++j) {
+            float ux = unitVerticesLines[k++];
+            float uy = unitVerticesLines[k++];
+
+            // Base
+            _vertexArray.push_back(ux * _radius); // x
+            _vertexArray.push_back(uy * _radius); // y
+            _vertexArray.push_back(0.f);          // z
+
+            // Top
+            _vertexArray.push_back(ux * _radius); // x
+            _vertexArray.push_back(uy * _radius); // y
+            _vertexArray.push_back(_length);      // z
+        }
+    }
+
+    // Indices for Base shape
+    for (uint8_t i = 0; i < _nShapeSegments; ++i) {
```

Comment:
I know you're setting th emax value of  `_nShapeSegments` to 32 but just a `ghoul_assert(_nShapeSegments.value <= std::numeric_limit<uint8_t>::max(), "Too many shape segments")` might ease some future pain.   If _nShapeSegments > 255, who knows what will happen /shrug